src/models/transaction/transaction.rs

```diff
@@ -109,17 +111,25 @@ impl StarknetTransaction {
         let max_priority_fee_per_gas = Some(client.max_priority_fee_per_gas());
 
         let calls: Calls = self.calldata()?.try_into()?;
-        let tx: TransactionSigned = (&calls).try_into()?;
+
+        if calls.len() > 1 {
+            return Err(EthApiError::ConversionError("Multi-call is not yet supported".to_string()));
+        }
```

Comment:
we could check here that `calls.len() != 1` and return an error otherwise.

Reply:
it's because it allows for len == 0 to error out, but if the len is 0, we're not in a multicall situation so it'll be a misleading error

Reply:
but we can change the error message to something like "Call length is {calls.len()}, expected 1".

Reply:
Also it depends a little on the phrasing but multi-call of length one is still multi-call no?